### PR TITLE
feat: global spinner config moved to registry

### PR DESCRIPTION
**Pull request by StyleT**

### CODE REVIEW

File: ilc/client/TransactionManager.js [hunk 1]
```diff
@@ -136,20 +136,19 @@ export class TransactionManager {
...
     #runGlobalSpinner = () => {
-        const registrySettings = this.#registrySettings;
-        if (registrySettings.globalSpinner === false || this.#spinnerTimeout) {
+        if (this.#spinnerConfig === false || this.#spinnerTimeout) {
             return;
         }
 
         this.#spinnerTimeout = setTimeout(() => {
-            if (registrySettings.globalSpinner === true) {
+            if (this.#spinnerConfig === true) {
```
Comment (nightnei) [inline ilc/client/TransactionManager.js:144]: do we have any ability to set boolean to spinnerConfig?

**Review by nightnei — CHANGES_REQUESTED** [review]

File: ilc/client/TransactionManager.js (hunk 1 above)
Comment (StyleT) [reply ilc/client/TransactionManager.js:144]: Not sure that I understood you... pls elaborate

Comment (nightnei) [reply ilc/client/TransactionManager.js:144]: sorry, my mistake, I read incorrectly this condition https://github.com/namecheap/ilc/blob/feature/i18n/global_spinner_config_to_registry/ilc/client/TransactionManager.js#L23

**Review by nightnei — APPROVED** [review]

File: ilc/client/TransactionManager.js [hunk 2]
```diff
@@ -11,18 +11,18 @@ export const slotWillBe = {
...
 
     #fakeSlots = [];
     #hiddenSlots = [];
     #transactionBlockers = [];
 
-    constructor(registrySettings = {}) {
-        this.#registrySettings = Object.assign({
-            globalSpinner: true
-        }, registrySettings);
+    constructor(registrySettings = {globalSpinner: {enabled: true, customHTML: ''}}) {
```
Comment (oleh-momot) [inline ilc/client/TransactionManager.js:22]: I am not sure that we should provide all settings here

File: ilc/client/TransactionManager.js [hunk 3]
```diff
@@ -11,18 +11,18 @@ export const slotWillBe = {
...
+        this.#spinnerConfig = !registrySettings.globalSpinner.customHTML || registrySettings.globalSpinner.enabled === false
```
Comment (oleh-momot) [inline ilc/client/TransactionManager.js:23]: Why do we need to have this difficult statement if we can just save settings?

File: registry/tests/settings.spec.ts [hunk 4]
```diff
@@ -11,7 +11,12 @@ describe(url, () => {
         it('should return settings and exclude values from secret records', async () => {
             const response = await request.get('/api/v1/settings').expect(200);
 
-            chai.expect(response.body).to.have.length(10);
+            const returnedKVs = (response.body as any[]).reduce((acc, v) => {
+                acc[v.key] = v.value;
+                return acc;
+            }, {});
+            chai.expect(returnedKVs['auth.openid.clientSecret']).to.be.undefined;
```
Comment (oleh-momot) [inline registry/tests/settings.spec.ts:18]: Please use the key from SettingKeys

File: registry/server/migrations/20201027180112_settings_globalSpinner.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import * as Knex from "knex";
+
+
+export async function up(knex: Knex): Promise<any> {
+    await knex('settings').insert([{
+        key: 'globalSpinner.enabled',
```
Comment (oleh-momot) [inline registry/server/migrations/20201027180112_settings_globalSpinner.ts:6]: I guess we should use here keys from SettingKeys and the same about Scopes and SettingTypes

**Review by oleh-momot — CHANGES_REQUESTED** [review]

File: registry/server/migrations/20201027180112_settings_globalSpinner.ts (hunk 5 above)
Comment (StyleT) [reply registry/server/migrations/20201027180112_settings_globalSpinner.ts]: fixed

File: registry/tests/settings.spec.ts (hunk 4 above)
Comment (StyleT) [reply registry/tests/settings.spec.ts]: fixed

File: ilc/client/TransactionManager.js (hunk 3 above)
Comment (StyleT) [reply ilc/client/TransactionManager.js]: fixed

File: ilc/client/TransactionManager.js (hunk 2 above)
Comment (StyleT) [reply ilc/client/TransactionManager.js]: fixed

**Review by oleh-momot — APPROVED** [review]

**Pull request merged**